tests/wpunit/EventTest.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+use StellarWP\Telemetry\Config;
+use StellarWP\Telemetry\Data_Providers\Null_Data_Provider;
+use StellarWP\Telemetry\Events\Event;
+use StellarWP\Telemetry\Opt_In\Status;
+use StellarWP\Telemetry\Telemetry\Telemetry;
+use StellarWP\Telemetry\Tests\Support\Traits\With_Test_Container;
+use StellarWP\Telemetry\Tests\Support\Traits\With_Uopz;
+
+class EventTest extends \Codeception\TestCase\WPTestCase {
+
+	use With_Test_Container;
+	use With_Uopz;
+
+	/**
+	 * It should successfully send event.
+	 *
+	 * @test
+	 */
+	public function should_send_event(): void {
+		$mock_response = [
+			'headers'       => [],
+			'body'          => json_encode( // phpcs:ignore WordPress.WP.AlternativeFunctions.json_encode_json_encode
+				[
+					'status'  => 'true',
+					'message' => 'success',
+				]
+			),
+			'response'      => [
+				'code'    => false,
+				'message' => false,
+			],
+			'cookies'       => [],
+			'http_response' => null,
+		];
+		$call_url      = null;
+		$call_args     = null;
+		$this->set_fn_return(
+			'wp_remote_post',
+			static function ( string $url, array $args ) use ( $mock_response, &$call_url, &$call_args ) {
+				$call_url  = $url;
+				$call_args = $args;
+
+				return $mock_response;
+			},
+			true
+		);
```

Comment:
I could see this being used by many of our tests. Should it be refactored into a function added at setup so you don't have to add it to each test?

Reply:
I wish... I spent some time working through that and there isn't a good way to store the `call_url` and `call_args` because the function passed to `$this->set_fn_return` runs statically and can't access the object running the test.

Unless you know of a way to access those, I'm going to leave this within the test.